### PR TITLE
MCKIN-15673 Japanese translations corrections

### PR DESCRIPTION
**Pull request by musmanmalik**
Japanese translations corrections

### CODE REVIEW

File: problem_builder/step.py [hunk 1]
```diff
@@ -251,8 +251,13 @@ def resource_string(path):
 
     def get_translation_content(self):
         try:
+            language = utils.translation.get_language().split('-')
+            if len(language) == 2:
+                new_lang = language[0] + "_" + language[1].upper()
+            else:
+                new_lang = language
             return self.resource_string('public/js/translations/{lang}/textjs.js'.format(
-                lang=utils.translation.get_language(),
```
Comment (viadanna) [inline problem_builder/step.py:255]: Didn't this work for Japanese? Why was it removed in favor of a manual approach?

File: problem_builder/public/js/translations/ja_JP/textjs.js [hunk 2]
```diff
@@ -29,7 +29,11 @@
     "Results retrieved on <%= creation_time %> (<%= seconds %> second).": [
       "\u7d50\u679c\u3092<%= creation_time %> (<%= seconds %> \u79d2) \u3067\u53d6\u5f97\u3057\u307e\u3057\u305f\u3002"
     ], 
-    "The report is currently being generated\u2026": "\u73fe\u5728\u30ec\u30dd\u30fc\u30c8\u3092\u4f5c\u6210\u4e2d\u3067\u3059\u2026"
+    "The report is currently being generated\u2026": "\u73fe\u5728\u30ec\u30dd\u30fc\u30c8\u3092\u4f5c\u6210\u4e2d\u3067\u3059\u2026",
```
Comment (viadanna) [inline problem_builder/public/js/translations/ja_JP/textjs.js:32]: While I can test this and review it works, I cannot attest the translation.

File: problem_builder/step.py (hunk 1 above)
Comment (musmanmalik) [reply problem_builder/step.py:255]: Here we are receiving language code like ja-jp, de-DE while we have repos ja_JP ,de_DE , etc. If we don't do like this the default en is returning. That is we need to do this.

Comment (musmanmalik) [reply problem_builder/step.py:255]: So for four character language code we need to change last 3 characters. `-` to `_` last two characters to upper case e.g. `jp` to `JP`.

File: problem_builder/public/js/translations/ja_JP/textjs.js (hunk 2 above)
Comment (musmanmalik) [reply problem_builder/public/js/translations/ja_JP/textjs.js:32]: We got these translations from trusted translator.

File: problem_builder/step.py (hunk 1 above)
Comment (bradenmacdonald) [reply problem_builder/step.py:255]: Might be good to have a comment here to make that more clear.

Comment (musmanmalik) [reply problem_builder/step.py:255]: @bradenmacdonald  Thank you for highlighting. I have added a comment. Please proceed.
Thanks

File: problem_builder/public/js/translations/ja_JP/textjs.js [hunk 3]
```diff
@@ -29,7 +29,11 @@
...
+    "You have used {num_used} of 1 submission.": [
+      "上限{max_attempts}回のうち {num_used} 回提出しました。",
+      "上限1回のうち {num_used} 回提出しました。"
+    ],
```
Comment (viadanna) [inline problem_builder/public/js/translations/ja_JP/textjs.js:36]: Not sure I understand correctly, but I think you have plural first, when it's expected to be second, e.g. https://github.com/open-craft/problem-builder/blob/d024a6ce2024adfe07facb72822aed16137d441f/problem_builder/public/js/translations/fr_CA/textjs.js#L34-L36

**Comment by musmanmalik** [comment]
@bradenmacdonald @viadanna  Please review this. A test is failing which seems not related to the changes in this pr. This may be a flaky test case. Rest if all is good kindly merge this and create a release tag. 

Thanks

File: problem_builder/public/js/translations/ja_JP/textjs.js (hunk 3 above)
Comment (musmanmalik) [reply problem_builder/public/js/translations/ja_JP/textjs.js:36]: @viadanna  have you tested this one? I have tested this and working fine as written in this pr.

Comment (viadanna) [reply problem_builder/public/js/translations/ja_JP/textjs.js:36]: image
This was the result for a setup with 3 attempts.

**Comment by viadanna** [comment]
Alright, merged and tagged, @musmanmalik

File: problem_builder/public/js/translations/ja_JP/textjs.js (hunk 3 above)
Comment (musmanmalik) [reply problem_builder/public/js/translations/ja_JP/textjs.js:36]: This is result is after first attempt out of three

Comment (musmanmalik) [reply problem_builder/public/js/translations/ja_JP/textjs.js:36]: `"You have used 1 of 3 submissions" needs to translate to "上限3回のうち１回提出しました。"`

Translator has given us the same

**Pull request merged**